x-pack/elastic-agent/pkg/agent/application/periodic.go

```diff
@@ -31,10 +31,12 @@ func (p *periodic) Start() error {
 
 	WORK:
 		for {
+			t := time.NewTimer(p.period)
 			select {
 			case <-p.done:
+				t.Stop()
```

Comment:
I think using `defer t.Stop()` below the `t := time.NewTimer(p.period)` is more future proof and more golang like. That way if a new case is ever added to the select then `t.Stop()` will already be handled.

Reply:
i tried to use defer whenever possible but i avoided this in a for loop.
as a follwup i can do func breakdown as in elastic-agent-client and extract loop body in a func which will be cleaner and allow using of defer statements
in this PR as this is blocker for 7.14 i want change to be as little intrusive as possible